packages/client/tests/functional/_utils/getTestSuitePlan.ts

```diff
@@ -41,11 +41,28 @@ export function getTestSuitePlan(
     config.matrixOptions.engineType ??= testCliMeta.engineType
   })
 
-  return expandedSuiteConfigs.map((namedConfig, configIndex) => ({
+  const testPlanEntries = expandedSuiteConfigs.map((namedConfig, configIndex) => ({
     name: getTestSuiteFullName(suiteMeta, namedConfig),
     skip: shouldSkipSuiteConfig(context, namedConfig, configIndex, testCliMeta, options),
     suiteConfig: namedConfig,
   }))
+
+  // when updating inline snapshots the first non-skipped test will perform the
+  // inline snapshot. the other tests are skipped otherwise jest will fail with
+  // "Multiple inline snapshots for the same call are not supported" error.
+  if (context.updateSnapshots === 'inline') {
+    let testWillPerformSnapshot = false
+    return testPlanEntries.reduce((acc, { name, skip, suiteConfig }) => {
+      if (skip === false && testWillPerformSnapshot === false) {
+        testWillPerformSnapshot = true
+        return [...acc, { name, skip, suiteConfig }]
+      } else {
+        return [...acc, { name, skip: true, suiteConfig }]
+      }
+    }, [] as TestPlanEntry[])
+  }
+
+  return testPlanEntries
```

Comment:
Related to the code deleted below. I found another bug that prevented me from updating many inline snapshots at once (for a single provider this time). The previous code had the assumption that the first test suite to pass would update the inline snapshots. In practice it is possible that the first test suite could also be skipped, and no snapshot would be updated. The previous code did not account for that.

The new code makes sure that the first non-skipped test suite found will be the one performing the inline snapshot update, and the rest will be skipped (as mentioned before: jest cannot allow parallel test suites to update a common inline snapshot).

Reply:
Nice, stumbled upon this couple of times but never took time for investigating.